### ISSUE TITLE
Bump HSQLDB from 2.5.0 to 2.6.1

### ISSUE THREAD

**Issue by tesshucom** [tesshucom/jpsonic#1145]
#### Summary

HSQLDB is bumped from 2.5.0 to 2.6.1.

It will be fixed an issue (regression) in some server shutdown cases. And other minor bugs will be fixed.

<details>
<summary>Caution (Users prior to v109.0.0 only)</summary>

 - Data conversion from 1.8.x is possible with 2.5.0. HSQLDB's migration implementation is not supported in later versions.
 - HSQLDB 1.8.x data was used by Jpsonic up to v109.0.0.
 - Jpsonic hasn't updated HSQLDB to 2.5.0 or later in v110.x.0. HSQLDB 2.5.0 has been used for about a year and a half.
 - Users prior to v109.0.0 can convert data by updating once from v109.1.0 to v110.x.x that Includes converter implementation. After starting once, stop the server, after which v111.xx or later will be available.

#### v109.0.0 - HSQLDB 1.8.x

Final version with HSQLDB 1.8.x data.

#### v110.x.x - HSQLDB 2.5.0

The final version of v110.x.x is stored on the release page.

#### v111.x.x - HSQLDB 2.6.0

Users using v109.1.0 or later can update without doing anything.

</details> 
<details>
<summary>Fixes</summary>

 - [x] Resize binary columns
     - If the size of a binary column is unspecified, it behaved as a variable region in previous versions, but not in 2.6.1.  The insert is rejected because it exceeds the size allowance, so it is changed to specify explicitly. Specifically, the target is the avatar image.
     - Jpsonic is basically a policy to add legacy database change logs without editing them. However, there are changes to the legacy schema set for two reasons.
         - In jpsonic-main/src/main/resources/liquibase/legacy/schema35.xml. Schema management for binary columns is a bit special. Cumulative logs for binary columns were created primarily by the multiple maintainers responsible for the Airsonic database. This issues fix can be seen as one of a series of changes to these. Based on these histories, it is appropriate to add to the same place.
         - The column changes must be completed before the first binary insert(A case where the database is built from an empty state at the time of new installation.). The first binary insert occurs in schema35.xml.
         - Changes are made only to hsqldb through Liquibase. So if you are using jndi, it has no effect at all.
 - [x] Remove data converter.
     - The 2.5.0 data converter implementation from 1.8x is based on the data converter implementation that exists in HSQLDB 2.5.0. The compatibility implementation that exists in HSQLDB 2.5.0 has been removed in 2.5.1.
     - Therefore, if 1.x data is read in 2.5.1 or later, it will be clearly treated as an error. `org.hsqldb.HsqlException: wrong database file version: unsupported old database version`
     - Airsonic's implementation of the 2.5.0 data converter from 1.8x can be left untouched. However, it will not be used at all in versions after 2.5.1.
     - Unnecessary code removed to simplify implementation
     - Instead, the final release version of v110.x.x, including Airsonic's data converter implementation, is always kept available for download from the release page.
 - [x] Add warning message against the data for HSQLDB 1.x(Users using v109.1.0 or later )
     - ``This version of Jpsonic does not support conversion from 1.x to 2.x. Please get https://github.com/jpsonic/jpsonic/releases/tag/v110.2.0 and upgrade your data of database.`` 

</details> 

#### Usage by the user

If you are using a version after v109.1.0, there are no special notes. Update the war file of the main body as usual. HSQLDB 2.6.1 will be available from the next boot.

If you are updating from a version earlier than v109.1.0, you cannot directly update to v111.x. Please update to v110.x once and then update to v111.x.

**Comment by tesshucom** [comment]
- The transition from 2.5.0 to 2.6.x was thought to be easy, but there seemed to be one annoying problem. sql.enforce_strict_size is more strict and the default seems to be true. In short, it's a field size issue.

> In connection with the above, an old database that did not have the enforce_strict_size property, is now converted to version 2.x with the engine supplying the missing size parameters. For example, a VARCHAR column declaration that has no size, is given a 32K size, a LONGVARCHAR column is given a 16MB size. Check these sizes are adequate for your use, and change the column definition as necessary.

- A surely observable phenomenon is that the avatar image insert fails. 
This is because what was a lengthable field because the field size was not set is blocked by the default size due to the lib update, causing truncation and causing an error. 
 - Earlier on Airsonic, there was a similar report on the path string. In Database running more Strict, the problem of not being able to scan directories that are too deep in the hierarchy. It's a rare case, but it suggests that it's not an avatar field-specific issue.
- There are two approaches: change enforce_strict_size to ignore, or size a specific field. In general, redesigning the size is rigorous and desirable. On the other hand, if you are already using an external DB with these Diarects in mind, adding a new design may produce bad results. It is less risky to maintain the status quo.



Whatever our policy in the end, we need to change the behavior of liquibase. Because, unlike the path string, the insert command that registers the avatar binary as default data occurs during the construction of the initial DB. (If we want to rewrite it, we need to rewrite the column definition after running liquibase with non-strict? ) This issue occurs when creating a new DB instead of an existing DB.

For the time being, it seems essential to operate enforce_strict_size with false in order to perform verification.

**Comment by tesshucom** [comment]
At the release note level, there are no updates that seem to have a specific impact.

topic

(2.5.1)

 - changed behaviour of WITH RECURSIVE with UNION ALL for compatibility with other systems -> UNION is used, but RECURSIVE is not used
 - fixed an issue (regression) in some server shutdown cases

Other minor bugs have been fixed by 2.6.1. Even if we postpone the design for "enforce_strict_size", there seems to be a benefit to updating.

**Comment by tesshucom** [comment]
The message seems to be wrong. No warning required when upgrading from version 2.5.0 to 2.6. There seems to be no problem with other processing.
